src/Provider.tsx

```diff
@@ -136,11 +136,16 @@ export function reducer(state: State, action: Action): State {
         return state;
       }
       const isGrid = currentTabStop.rowIndex !== null;
+      const isFirst = index === 0;
+      const isLast = index === state.tabStops.length - 1;
       const navigation = getNavigationValue(
         key,
         ctrlKey,
         isGrid,
-        state.direction
+        state.direction,
+        !!state.shouldLoopAround,
```

Comment:
If `shouldLoopAround` is made not optional as per my other comment then there is no need for a double-bang here.